webfrontend/src/components/setup/setup-tags.tsx

```diff
@@ -0,0 +1,212 @@
+    fetch('/profile/obtain_tags')
+        .then(response => {
+            if (response.status !== 200) {
+                triggerErrorMessage();
+            } else {
+                response.json()
+                    .then(response => {
+                        setTags(response.tags);
+                        setTagCountLimit(response.tag_count_limit);
+                       
+                    });
+            }
+        });
+    
+    fetch('/profile/can_remove_tag')
+        .then(response => {
+            if (response.status !== 200) {
+                triggerErrorMessage();
+                return;
+            }
+            response.text().then(text => {
+                setCanRemoveTag(text === 'true');
+            })
+        })
+    }, [tags]);
+useEffect(() => {
+  document.addEventListener('click', (event: any) => {
+      if (searchTagForm.current?.contains(event.target)) {
+          setShowTagResult(true);
+      } else {
+          setShowTagResult(false);
+      }
+  })
+    }, []);
+  const submitTags = useCallback((event: React.SyntheticEvent<EventTarget>) => {
+    event.preventDefault();
+    if (!isLoading) {
+        setIsLoading(true);
+        fetch('/profile/add_tags', postRequestContent({
+            count: toBeSubmitted.length,
+            tags: toBeSubmitted.map(tag => tag.name)
+        }))
+            .then(response => {
+                setIsLoading(false);
+                if (response.status !== 200) {
+                    triggerErrorMessage();
+                } else {
+                    setTags([...tags].concat(toBeSubmitted));
+                    setToBeSubmitted([]);
+                    setShowAlert(false);
+                }
+            });
+    }
+  }, [isLoading, tags, toBeSubmitted]);
+  const searchTag = useCallback((event: React.SyntheticEvent<EventTarget>) => {
+    event.preventDefault();
+    fetch("/profile/search_tags?tag=" + searchParam)
+        .then(response => {
+            if (response.status !== 200) {
+                triggerErrorMessage();
+            } else {
+                response.json().then(response => {
+                    setSearchResults(response.tags.filter((tag:Tag) => toBeSubmitted.find(addedTag => addedTag.name === tag.name) === undefined));
+                    setSearchDone(true);
+                })
+            }
+        })
+  },[searchParam, toBeSubmitted]);
+  const addNewTag = useCallback((index: number) => {
+    if (tags.length + toBeSubmitted.length < tagCountLimit) {
+        setSearchResults(searchResults.filter((_, i) => i !== index));
+        setToBeSubmitted([...toBeSubmitted, searchResults[index]]);
+    } else {
+        addTagMessageButton.current?.click();
+    }
+}, [searchResults, tagCountLimit, tags.length, toBeSubmitted]);
+
+const removeNewTag = useCallback((index: number) => {
+    setToBeSubmitted(toBeSubmitted.filter((_, i) => i !== index));
+  }, [toBeSubmitted]);
+
+const removeTag = useCallback(() => {
```

Comment:
Poor indentation. `const` declarations should be indented. Please keep your code properly indented so that readers know which line belongs to a block and which does not.